RacingCar/Calculator.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Calculator.swift
+//  RacingCar
+//
+//  Created by main on 2022/05/02.
+//
+
+import Foundation
+
+enum CalculatorError: Error {
+    case divideByZero
+    case emptyString
+    case unknownOperator
+}
+
+class Calculator {
+    
+    private var evaluation: Int?
+    private var currentOperator: String?
+    
+    private func add(lhs: Int, rhs: Int) -> Int {
+        return lhs + rhs
+    }
+    
+    private func sub(lhs: Int, rhs: Int) -> Int {
+        return lhs - rhs
+    }
+    
+    private func mul(lhs: Int, rhs: Int) -> Int {
+        return lhs * rhs
+    }
+    
+    private func div(lhs: Int, rhs: Int) throws -> Int {
+        
+        if rhs == 0 {
+            throw CalculatorError.divideByZero
+        }
+        
+        return lhs / rhs
+    }
+    
+    private func calculate(lhs: Int, rhs: Int, op: String) throws -> Int {
+        switch op {
+        case "+":
+            return add(lhs: lhs, rhs: rhs)
+        case "-":
+            return sub(lhs: lhs, rhs: rhs)
+        case "*":
+            return mul(lhs: lhs, rhs: rhs)
+        case "/":
+            return try div(lhs: lhs, rhs: rhs)
+        default:
+            throw CalculatorError.unknownOperator
+        }
+    }
+    
+    private func isInteger(_ input: String) -> Bool {
+        if Int(input) != nil {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    func evaluate(input: String) throws -> Int {
+        if input.isEmpty {
+            throw CalculatorError.emptyString
+        }
+        
+        let tokens: [String] = input.components(separatedBy: " ")
+        
+        let operators = tokens.filter { !isInteger($0) }
```

Comment:
가독성을 위해서 아래처럼 바꿔볼 수도 있습니다.

```suggestion
        let operators = tokens.filter { false == isInteger($0) }
```

Reply:
보통 저는 이런코를 팀의 코딩 스타일을 보고 따라하는편이었던것같습니다. 여러방식이 섞여있는게 가독성을 해치기도 하는것 같아서요.
그리고 비교문을 사용하더라도 false 를 왼쪽에 둘지 오른쪽에 둘지도 고민인데 보통은 오른쪽에 두더라구요... 

c언어를 사용하던 시절에 ==비교문을 작성하려다가 =를 하나 빠뜨리면 바로 애러를 일으켜서 알아채도록 false를 왼쪽에 두는 방식을 선호했다고 들어는 보았는데 아직 다른사람의 코드에서 한번도 그런코드를 본적은 없었던것같습니다. (이번이 처음일지도...)

!를 사용하는것 자체가 햇갈릴 위험이 많아지는 방식이어서 !를 안사용할 수 있으면 안사용하는 방식으로, 즉 제안주신 코드가 가장 적절하다고 생각됩니다.
저부터 이런 스타일로 코드를 작성해보도록 노력해보겠습니다.